--- a/src/qtui/chatviewsettings.h
+++ b/src/qtui/chatviewsettings.h
@@ -104,7 +104,7 @@ public:
      */
     inline void enableSenderBrackets(bool enabled) { setLocalValue("ShowSenderBrackets", enabled); }

-    inline QString webSearchUrlFormatString() { return localValue("WebSearchUrlFormat", "https://www.google.com/search?q=%s").toString(); }
+    inline QString webSearchUrlFormatString() { return localValue("WebSearchUrlFormat", "https://duckduckgo.com/?q=%s").toString(); }
     inline void setWebSearchUrlFormatString(const QString &format) { setLocalValue("WebSearchUrlFormat", format); }
 };

--- a/src/qtui/settingspages/chatviewsettingspage.ui
+++ b/src/qtui/settingspages/chatviewsettingspage.ui
@@ -244,7 +244,7 @@
         <string notr="true">WebSearchUrlFormat</string>
        </property>
        <property name="defaultValue" stdset="0">
-        <string>https://www.google.com/search?q=%s</string>
+        <string>https://duckduckgo.com/?q=%s</string>
        </property>
       </widget>
      </item>
