--- a/config.def.h
+++ b/config.def.h
@@ -1121,98 +1121,98 @@ static const bool enable_device_vibration    = false;
 #endif
 
 #if defined(HAKCHI)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://hakchicloud.com/Libretro_Cores/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(ANDROID)
 #if defined(ANDROID_ARM_V7)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/android/latest/armeabi-v7a/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(ANDROID_ARM)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/android/latest/armeabi/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(ANDROID_AARCH64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/android/latest/arm64-v8a/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(ANDROID_X86)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/android/latest/x86/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(ANDROID_X64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/android/latest/x86_64/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #else
 #define DEFAULT_BUILDBOT_SERVER_URL ""
 #endif
 #elif defined(__QNX__)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/blackberry/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(IOS)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/apple/ios/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(OSX)
 #if defined(__x86_64__)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/apple/osx/x86_64/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__i386__) || defined(__i486__) || defined(__i686__)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://bot.libretro.com/nightly/apple/osx/x86/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #else
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/apple/osx/ppc/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #endif
 #elif defined(_WIN32) && !defined(_XBOX)
 #if _MSC_VER >= 1910
 #ifndef __WINRT__
 #if defined(__x86_64__) || defined(_M_X64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2017-desktop/x64/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__i386__) || defined(__i486__) || defined(__i686__) || defined(_M_IX86) || defined(_M_IA64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2017-desktop/x86/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__arm__) || defined(_M_ARM)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2017-desktop/arm/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__aarch64__) || defined(_M_ARM64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2017-desktop/arm64/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #endif
 #else
 #if defined(__x86_64__) || defined(_M_X64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2017-uwp/x64/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__i386__) || defined(__i486__) || defined(__i686__) || defined(_M_IX86) || defined(_M_IA64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2017-uwp/x86/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__arm__) || defined(_M_ARM)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2017-uwp/arm/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__aarch64__) || defined(_M_ARM64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2017-uwp/arm64/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #endif
 #endif
 #elif _MSC_VER == 1600
 #if defined(__x86_64__) || defined(_M_X64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2010/x86_64/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__i386__) || defined(__i486__) || defined(__i686__) || defined(_M_IX86) || defined(_M_IA64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2010/x86/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #endif
 #elif _MSC_VER == 1400
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2005/x86/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif _MSC_VER == 1310
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows-msvc2003/x86/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #else
 #if defined(__x86_64__) || defined(_M_X64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows/x86_64/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__i386__) || defined(__i486__) || defined(__i686__) || defined(_M_IX86) || defined(_M_IA64)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/windows/x86/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #endif
 #endif
 #elif defined(__linux__)
 #if defined(__x86_64__)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/linux/x86_64/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__i386__) || defined(__i486__) || defined(__i686__)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/linux/x86/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__arm__) && __ARM_ARCH == 7 && defined(__ARM_PCS_VFP)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/linux/armhf/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #else
 #define DEFAULT_BUILDBOT_SERVER_URL ""
 #endif
 #elif defined(WIIU)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/nintendo/wiiu/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(HAVE_LIBNX)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://buildbot.libretro.com/nightly/nintendo/switch/libnx/latest/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__CELLOS_LV2__) && defined(DEX_BUILD)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://libretro.xbins.org/libretro/nightly/playstation/ps3/latest/dex-ps3/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__CELLOS_LV2__) && defined(CEX_BUILD)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://libretro.xbins.org/libretro/nightly/playstation/ps3/latest/cex-ps3/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #elif defined(__CELLOS_LV2__) && defined(ODE_BUILD)
-#define DEFAULT_BUILDBOT_SERVER_URL "http://libretro.xbins.org/libretro/nightly/playstation/ps3/latest/ode-ps3/"
+#define DEFAULT_BUILDBOT_SERVER_URL ""
 #else
 #define DEFAULT_BUILDBOT_SERVER_URL ""
 #endif
 
-#define DEFAULT_BUILDBOT_ASSETS_SERVER_URL "http://buildbot.libretro.com/assets/"
+#define DEFAULT_BUILDBOT_ASSETS_SERVER_URL ""
 
 #define DEFAULT_DISCORD_APP_ID "475456035851599874"
 
--- a/frontend/drivers/platform_darwin.m
+++ b/frontend/drivers/platform_darwin.m
@@ -459,7 +459,7 @@ static void frontend_darwin_get_environment_settings(int *argc, char *argv[],
     char assets_zip_path[PATH_MAX_LENGTH];
 #if TARGET_OS_IOS
     if (major > 8)
-       strlcpy(g_defaults.path.buildbot_server_url, "http://buildbot.libretro.com/nightly/apple/ios9/latest/", sizeof(g_defaults.path.buildbot_server_url));
+       strlcpy(g_defaults.path.buildbot_server_url, "", sizeof(g_defaults.path.buildbot_server_url));
 #endif
 
     fill_pathname_join(assets_zip_path, bundle_path_buf, "assets.zip", sizeof(assets_zip_path));
--- a/libretro-common/samples/net/net_http_test.c
+++ b/libretro-common/samples/net/net_http_test.c
@@ -37,7 +37,7 @@ int main(void)
    if (!network_init())
       return -1;
 
-   http1 = net_http_new("http://buildbot.libretro.com/nightly/windows/x86_64/latest/mednafen_psx_libretro.dll.zip");
+   http1 = net_http_new("");
 
    while (!net_http_update(http1, &pos, &tot))
       printf("%.9lu / %.9lu        \r",pos,tot);
--- a/retroarch.cfg
+++ b/retroarch.cfg
@@ -721,13 +721,13 @@
 #### Core Updater
 
 # URL to core update directory on buildbot.
-# core_updater_buildbot_url = "http://buildbot.libretro.com"
+core_updater_buildbot_url = ""
 
 # URL to assets update directory on buildbot.
-# core_updater_buildbot_assets_url = "http://buildbot.libretro.com/assets/"
+core_updater_buildbot_assets_url = ""
 
 # After downloading, automatically extract archives that the downloads are contained inside.
-# core_updater_auto_extract_archive = true
+core_updater_auto_extract_archive = false
 
 #### Network
 
