--- bibletime-2.11.2.orig/CMakeLists.txt	2018-06-18 15:07:00.000000000 -0500
+++ bibletime-2.11.2/CMakeLists.txt	2020-04-10 12:50:57.857672342 -0500
@@ -156,21 +156,10 @@
     ADD_DEFINITIONS("-DBT_MOBILE")
 ENDIF()
 
-FIND_PACKAGE(Qt5WebEngine QUIET)
-IF(NOT Qt5WebEngine_FOUND)
     MESSAGE(STATUS "Using Qt WebKit")
     FIND_PACKAGE(Qt5WebKit REQUIRED)
     FIND_PACKAGE(Qt5WebKitWidgets REQUIRED)
     SET(USEWEBENGINE OFF)
-ELSE()
-    MESSAGE(STATUS "Using Qt WebEngine")
-    FIND_PACKAGE(Qt5WebEngineWidgets REQUIRED)
-    SET(USEWEBENGINE ON)
-ENDIF()
-
-IF(USEWEBENGINE)
-    ADD_DEFINITIONS("-DUSEWEBENGINE")
-ENDIF()
 
 # Define $bibletime_SOURCES
 INCLUDE("${CMAKE_CURRENT_SOURCE_DIR}/cmake/BTSourceFileList.cmake")
@@ -232,8 +221,6 @@
 
 IF(${BIBLETIME_FRONTEND} STREQUAL "MOBILE")
   TARGET_LINK_LIBRARIES("bibletime" Qt5::Widgets Qt5::WebKitWidgets Qt5::Qml Qt5::Quick Qt5::Xml Qt5::Network Qt5::PrintSupport Qt5::Svg)
-ELSEIF(USEWEBENGINE)
-  TARGET_LINK_LIBRARIES("bibletime" Qt5::Widgets Qt5::WebEngine Qt5::WebEngineWidgets Qt5::WebChannel Qt5::Xml Qt5::Network Qt5::PrintSupport Qt5::Svg)
 ELSE()
   TARGET_LINK_LIBRARIES("bibletime" Qt5::Widgets Qt5::WebKit Qt5::WebKitWidgets Qt5::Xml Qt5::Network Qt5::PrintSupport Qt5::Svg)
 ENDIF()
@@ -288,13 +275,8 @@
          DESTINATION "${BT_SHARE_PATH}/bibletime/display-templates/")
 
 
-IF(USEWEBENGINE)
-    INSTALL(FILES "src/frontend/display/btwebengine.js"
-     DESTINATION "${BT_SHARE_PATH}/bibletime/javascript/")
-ELSE()
-    INSTALL(FILES "src/frontend/display/bthtml.js"
-        DESTINATION "${BT_SHARE_PATH}/bibletime/javascript/")
-ENDIF()
+INSTALL(FILES "src/frontend/display/bthtml.js"
+    DESTINATION "${BT_SHARE_PATH}/bibletime/javascript/")
 
 INSTALL(FILES "docs/license.html"
         DESTINATION "${BT_SHARE_PATH}/bibletime/license/")